src/logging.h

```diff
@@ -68,6 +69,121 @@ namespace BCLog {
         BLOCKSTORE  = (1 << 26),
         ALL         = ~(uint32_t)0,
     };
+
+
+    constexpr auto escape_sequence{"\033["};
+
+    class Color
```

Comment:
`logging.h` is widely included, so it may be best to move as much new code into the logging.cpp implementation file instead. 

Also, there are refactorings to come in #25203 to de-duplicate and maybe speed up some some of the logging.cpp code, along with benchmarks for it; it may be a good idea to test this rebased on that pull once it is updated and see how the new code performs in the benchmarks.

Reply:
I will change this but considering there are some Concept NACKS on this PR I'm not sure I should work on this yet.

Reply:
Yes, I don't know. Here is the pull that added colors to -getinfo, if the discussions there are helpful for you: #21832. Of course, -getinfo is much more isolated code than the logging.